src/Libraries/Nop.Data/Mapping/Builders/Customers/CustomerBuilder.cs

```diff
@@ -23,7 +25,24 @@ public override void MapEntity(CreateTableExpressionBuilder table)
                 .WithColumn(nameof(Customer.Username)).AsString(1000).Nullable()
                 .WithColumn(nameof(Customer.Email)).AsString(1000).Nullable()
                 .WithColumn(nameof(Customer.EmailToRevalidate)).AsString(1000).Nullable()
+                .WithColumn(nameof(Customer.FirstName)).AsString(1000).Nullable()
+                .WithColumn(nameof(Customer.LastName)).AsString(1000).Nullable()
+                .WithColumn(nameof(Customer.Gender)).AsString(1000).Nullable()
+                .WithColumn(nameof(Customer.Company)).AsString(1000).Nullable()
+                .WithColumn(nameof(Customer.StreetAddress)).AsString(1000).Nullable()
+                .WithColumn(nameof(Customer.StreetAddress2)).AsString(1000).Nullable()
+                .WithColumn(nameof(Customer.ZipPostalCode)).AsString(1000).Nullable()
+                .WithColumn(nameof(Customer.City)).AsString(1000).Nullable()
+                .WithColumn(nameof(Customer.County)).AsString(1000).Nullable()
+                .WithColumn(nameof(Customer.Phone)).AsString(1000).Nullable()
+                .WithColumn(nameof(Customer.Fax)).AsString(1000).Nullable()
+                .WithColumn(nameof(Customer.VatNumber)).AsString(1000).Nullable()
+                .WithColumn(nameof(Customer.TimeZoneId)).AsString(1000).Nullable()
+                .WithColumn(nameof(Customer.CustomCustomerAttributesXML)).AsString(int.MaxValue).Nullable()
+                .WithColumn(nameof(Customer.DateOfBirth)).AsDateTime2().Nullable()
                 .WithColumn(nameof(Customer.SystemName)).AsString(400).Nullable()
+                .WithColumn(NameCompatibilityManager.GetColumnName(typeof(Customer), nameof(Customer.CurrencyId))).AsInt32().ForeignKey<Currency>(onDelete: Rule.SetNull).Nullable()
+                .WithColumn(NameCompatibilityManager.GetColumnName(typeof(Customer), nameof(Customer.LanguageId))).AsInt32().ForeignKey<Language>(onDelete: Rule.SetNull).Nullable()
```

Comment:
you don't need to use the NameCompatibilityManager for the new fields, since their names will certainly be correct in terms of the name. NameCompatibilityManager - used only for old fields in the database that were named through the _ symbol